addons/account/models/account_move.py

```diff
@@ -298,6 +302,23 @@ def _search_default_journal(self, journal_types):
     inalterable_hash = fields.Char(string="Inalterability Hash", readonly=True, copy=False)
     string_to_hash = fields.Char(compute='_compute_string_to_hash', readonly=True)
 
+    # -------------------------------------------------------------------------
+    # HELPERS
+    # -------------------------------------------------------------------------
+
+    def _convert_to_debit_credit(self, balance):
+        self.ensure_one()
+        if self.use_storno_accounting:
+            return {
+                'debit': balance if balance < 0.0 else 0.0,
+                'credit': -balance if balance > 0.0 else 0.0,
+            }
+        else:
+            return {
+                'debit': balance if balance > 0.0 else 0.0,
+                'credit': -balance if balance < 0.0 else 0.0,
+            }
```

Comment:
nonsense. The method should receive debit and credit and inverse them if not storno and negative...

Reply:
We should receive the balance, and compute the debit/credit from there instead of the opposite IMO.
Then there is no need to have `use_storno_accounting` on the move either.

Is it needed anyways ? Why not check the type of the move and the company setting? It should be cached, so it doesn't impact the perfs, but we remove that confusing field on `account.move`

Reply:
@qdp-odoo This way is better. The only thing I want to improve is to give balance instead of debit/credit when creating an account.move.line. This is not already done in this branch because I wanted a first opinion before making this huge change everywhere.